CONTRIBUTING.md

```diff
@@ -24,19 +24,79 @@ Please follow these steps when making pull requests (PRs):
     Please give the PR a descriptive name.
 
 1.  As you make progress on your PR, keep your branch up-to-date with the `master` branch which
-        may have been updated after starting your PR.
+        may have been updated *after* starting your PR.
     Your PR **MUST** be updated to reflect changes to `master` in order to be merged.
-    Use the following procedure for updating your branch:
+    Use the following procedure for updating your branch and are ready to commit your changes:
 
     <!--- language: lang-shell -->
 
-        # While on your PR branch <issue-branch>
+        ## While on your PR branch <issue-branch> hosted at <your-remote> repository:
         git commit
         git checkout master
-        git pull origin --rebase
+        git pull <illixr-remote> --rebase                           ## Line A
         git checkout <issue-branch>
-        git rebase master
-        git push origin <issue-branch> --force-with-lease
+        git rebase master                                           ## Line B
+        git push <your-remote> <issue-branch> --force-with-lease    ## Line C
+
+    If you want to update your branch while you are still making changes to your branch:
+
+    <!--- language: lang-shell -->
+
+        ## While on your PR branch <issue-branch> hosted at <your-remote> repository:
+        git stash
+        git checkout master
+        git pull <illixr-remote> --rebase                           ## Line A
+        git checkout <issue-branch>
+        git rebase master                                           ## Line B
+        git stash apply <stash-number>
```

Comment:
Use `git stash pop` instead of git stash apply to also keep your stash clean. If popping fails for some reason, git will not drop your work.

Reply:
I used `git stash apply` here to not assume that changes from this branch/feature are at the top of the stash queue. `git stash drop <stash-number>` can then be used to clean up.